pycbc/events/coinc.py

```
@@ -245,92 +249,106 @@ def time_coincidence(t1, t2, window, slide_step=0):
-            #
-            # To modify that assumption, the code here would be modified
-            # by adding a function that remaps the coinc time frame and unmaps
-            # it and the end of this loop.
-            # This remapping must ensure
-            #    * function of the standard slide number
-            #    * ensure all times remain within coincident segment
-            #    * unbiased distribution of triggers after mapping.
-
+        # tsort gives ordering from original order to time sorted order
+        tsort = otime.argsort()
+        time1 = otime[tsort]
+
+        # Find coincidences between dependent ifo triggers and existing coincs
+        # cycle over both fixed and pivot
+        # at the 1st iteration, the fixed and pivot times and ids are reduced to
+        #  those for which the first (out of fixed/pivot) forms a coinc with ifo1
+        # at the 2nd iteration, we are left with triggers for which both fixed
+        #  and pivot are coincident with ifo1
+        for ifo2 in ids:
+            logging.info('added ifo %s, testing against %s' % (ifo1, ifo2))
             w = win(ifo1, ifo2)
-            left = numpy.searchsorted(time, ctimes[ifo2] - w)
-            right = numpy.searchsorted(time, ctimes[ifo2] + w)
-
-            # remove elements that will not form a coinc
-            # There is only at most one trigger for an existing coinc
-            # (assumes triggers spaced > slide step)
+            left = numpy.searchsorted(time1, ctimes[ifo2] - w)
+            right = numpy.searchsorted(time1, ctimes[ifo2] + w)
+            # any times within time1 coincident with the time in ifo2 have
+            # indices between left and right
+            # nz indexes into times in ifo2 which have coincidences with ifo1
+            # times
             nz = (right - left).nonzero()
+            if len(nz[0]) and (right - left).max() > 1:
+                # we expect at most one coincident time in ifo1, assuming
```

The 'easiest' solution here would just be to ignore one of these coincident triggers in ifo1 - ie silently proceed with just taking the first trigger, which I think left[nz] already does.